pkg/trace/agent/agent.go

```diff
@@ -372,6 +365,22 @@ func (a *Agent) Process(p *api.Payload) {
 	}
 }
 
+// processedTrace creates a ProcessedTrace based on the provided chunk and root.
+// It makes a deep copy of the provided chunk to ensure that any changes to the
+// chunk will not affect the TraceChunk of the ProcessedTrace.
```

Comment:
nit: `[...] to ensure that any subsequent changes made to the original chunk will not [...]`